sherpa/astro/ui/utils.py

```diff
@@ -650,6 +647,8 @@ def dataspace1d(self, start, stop, step=1, numbins=None,
 
         """
         # support non-integrated grids with inclusive boundaries
+        # We do NOT want to use an isinstance check since Data1DInt is
+        # derived from Data1D.
         if dstype in (sherpa.data.Data1D, sherpa.astro.data.DataPHA):
```

Comment:
@olaurino had a question about this in his work cleaning up the Data hierarchy in #614 but I'm  not sure we ever resolved this. Technically the PHA data is integrated, but the PHA standard/documentation kind-of treats it as channel, counts (or rate) rather than channel_lo, channel_hi, counts and our code has (I fell) implicitly treated it more like a Data1D rather than Data1DInt class (but not always).

If you look at DataPHA https://sherpa.readthedocs.io/en/latest/data/api/sherpa.astro.data.DataPHA.html you'll see that it derives from Data1D (probably unfortunately).